Competition2017/src/competition/subsystems/feeder/SideFeederSubsystem.java

```diff
@@ -0,0 +1,170 @@
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class SideFeederSubsystem extends BaseSubsystem {
+
+    private static Logger log = Logger.getLogger(FeederSubsystem.class);
+
+    private final XCANTalon masterMotor;
+
+    public enum FeederSide {
+
+        Left, 
+
+        Right
+    }
+
+    private final FeederSide side;
+    
+    private final DoubleProperty feederTestingPowerStep;
+    private final DoubleProperty feederCurrentSpeed;
+    private final DoubleProperty feederTargetSpeed;
+    private final DoubleProperty feederOutputPower;
+    private final DoubleProperty feederTalonError;
+    
+    private final BooleanProperty atSpeedProp;
+    private final BooleanProperty enableFeederLogging;
+
+    private final DoubleProperty p;
+    private final DoubleProperty i;
+    private final DoubleProperty d;
+    private final DoubleProperty f;
+
+    private final DoubleProperty feederSpeedThresh;
+
+    public SideFeederSubsystem(int motor, FeederSide side, WPIFactory factory, XPropertyManager propManager){
+
+        log.info("Creating " + side + " Feeder");
+
+        this.side = side;
+
+        p = propManager.createPersistentProperty(side + "Feeder wheel P", 0);
+        i = propManager.createPersistentProperty(side + "Feeder wheel I", 0);
+        d = propManager.createPersistentProperty(side + "Feeder wheel D", 0);
+        f = propManager.createPersistentProperty(side + "Feeder wheel F", 0);
+
+        feederTestingPowerStep = propManager.createPersistentProperty(side + "Feeder power step", 0.025);
+        feederSpeedThresh = propManager.createPersistentProperty(side + "Feeder nominal speed thresh (TPC)", motor);
+        feederCurrentSpeed = propManager.createEphemeralProperty(side + "Feeder current speed (TPC)", 0);
+        feederTargetSpeed = propManager.createEphemeralProperty(side + "Feeder goal speed (TPC)", 0);
+        feederOutputPower = propManager.createEphemeralProperty(side + "Feeder voltage", 0);
+
+        atSpeedProp = propManager.createEphemeralProperty("Is" + side + "Feeder at speed?", false);
+        feederTalonError = propManager.createEphemeralProperty(side + "Feeder speed error", 0);
+        enableFeederLogging = propManager.createEphemeralProperty("Is " + side + "Feeder logging info?", false);
+
+        this.masterMotor = factory.getCANTalonSpeedController(motor);
+
+        setUpMotorTeam(masterMotor);
+        masterMotor.createTelemetryProperties(side + "Feeder master", propManager);
+    }
+
+    private void setUpMotorTeam(XCANTalon master) {
+
+        log.info("Configuring " + side + "motor");
+
+        master.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+        master.reverseSensor(true);
+        master.setInverted(true);
+
+        log.info("Configuring voltages");
+
+        master.configNominalOutputVoltage(0,  -0);
+        master.configPeakOutputVoltage(12, -6);
+        master.setProfile(0);
+        master.setF(.099);
```

Comment:
set this to 0 for now - we won't know what the proper value is until we do some tests.